### PR TITLE
PR on base proposal by Jim

### PR DESCRIPTION
**Pull request by Zimmi48**

### CODE REVIEW

File: text/refactor_refman.md [hunk 1]
```diff
@@ -103,6 +167,8 @@ a good writer with a subject expert to write or update some sections.
 We will also consider providing with copy editing assistance beyond
```
Comment (jfehrle) [inline text/refactor_refman.md]:
```suggestion
We will also consider providing copy editing assistance beyond
```

File: text/refactor_refman.md [hunk 2]
```diff
@@ -17,9 +17,11 @@ contribute code enhancements and improvements to Coq.
...
 We believe the documentation should be easily understood by users who
 have only a bachelor's degree in Computer Science or Mathematics.
-It should be possible to learn and master Coq by reading the reference
-manual.  On the other hand, the reference manual shouldn't try to
-be a tutorial or a textbook.
+Users who are learning Coq should be able to use the manual as a
+reference and exhaustive documentation without having to become expert
+first. Existing online ressources should be enough to learn Coq, and
+the reference manual should be a good complement to one of the
+standard tutorials.
```
Comment (jfehrle) [inline text/refactor_refman.md]: How about something along these lines?  (as long as we drop "Existing online ressources should be enough to learn Coq")

Those who are learning Coq should be able to use the manual as an authoritative reference without becoming an expert first.  The manual complements existing tutorials; tutorials are good for a first exposure to Coq while the manual covers all the details.

File: text/refactor_refman.md [hunk 3]
```diff
@@ -35,19 +37,14 @@ Some of the more apparent issues with the current documentation are:
      scattered across multiple sections.
 -	**Hard to locate concepts in the document.** For example, there is
      no subject index where the reader could look up unification.  The
-     HTML search box is not very helpful. (You can't see the
-     matching text for each search result without clicking on the
-     link.  Clicking on the link takes you to the top of the section
-     rather than to the matching text.)
+     HTML search box is not very helpful. (This is something that will
```
Comment (jfehrle) [inline text/refactor_refman.md]: I think it would be better to keep the original detail (below).  It is relevant to usability, although I don't expect we'll get to this soon and perhaps not at all.  This section is the motivation.  It's not a promise that we address every point.

(You can't see the
     matching text for each search result without clicking on the
     link.  Clicking on the link takes you to the top of the section
     rather than to the matching text.)

File: text/refactor_refman.md [hunk 4]
```diff
@@ -65,10 +62,78 @@ Other areas for improvement include:
      are so long it's easy to lose context or wander into unrelated
      material.
 
-# Working Outline
-
-See *\<link to PR>.* The outline is subject to revision as the project
-proceeds.
+# New structure
```
Comment (jfehrle) [inline text/refactor_refman.md:67]: Let's make the outline a separate PR as we discussed.

File: text/refactor_refman.md [hunk 5]
```diff
@@ -112,7 +178,11 @@ https://github.com/coq/coq/pull/10614, for example, updates the
 
 # Disadvantages
 
-
+Working on refactoring the manual incrementally will create
+intermediate states which can be more confusing than what we started
+with.  If we end up not having finalized the refactoring towards the
+end of a release cycle, we will need to make adjustments so that it
+still looks good.
```
Comment (jfehrle) [inline text/refactor_refman.md]: >> Could be more positive:

>>  Improving the documentation incrementally may require additional effort to ensure that the documentation remains usable.

> Well, I don't want to diminish the drawbacks.

I think the first sentence is too strong.  Many of the changes can be done without hurting usability.

> Working on refactoring the manual incrementally will create
intermediate states which can be more confusing than what we started
with.

File: text/refactor_refman.md [hunk 6]
```diff
@@ -88,12 +153,11 @@ considerations:
...
-directory.  We'll use this outline to track the progress of the
-project and who's doing what.  As we begin working on each chapter
-and section, we'll begin by reviewing the current chapter in a PR
-to identify what changes are needed, followed shortly afterward by
-creating and reviewing a revised chapter or section outline.
+As we begin working on each chapter and section of the new outline,
+we'll begin by reviewing available material to identify what changes
+are needed, and what writing will be required.  We will leave
+placeholders asking for contributions for sections where significant
+writing is needed.
```
Comment (jfehrle) [inline text/refactor_refman.md]: You'll tweak this as discussed?

**Review by jfehrle — COMMENTED** [review]
A few comments.  We don't need to spend a lot of time wordsmithing this to perfection.  Hope we can submit this as a CEP and PR soon, at least by the first week in January.

**Comment by jfehrle** [comment]
After further thought, I'm more convinced refactoring the vernacular chapter first is the best plan.  I think every tutorial touches on this material early.  I think the tactics chapter might be a good second chapter to refactor and maybe Ltac third.  If these chapters need additional supporting material that belongs in other chapters, we should be able to add that material before we get to refactor those other chapters.

**Comment by Zimmi48** [comment]
@jfehrle Very sorry that I waited so long to update the CEP draft. Unfortunately, I don't remember everything that we said during our video-chat. On the other hand, there are some new thoughts on how to proceed with the refactoring and minimizing merge conflicts. I hope we can finish preparing this draft soon so that we can submit it to the others and proceed to merge the first refactoring PRs (besides the grammar ones).

**Comment by jfehrle** [comment]
No worries.  One of the ideas we discussed was to create chapter/section outlines as we begin working on those chapters/sections.  These could be done as PRs to facilitate discussion/review and preserved in the source tree.  Could just say we *may* do this--we'll try and see if it's actually helpful.

Alas, I can't use the review process to make suggestions.  So here are some done the old-fashioned way:

"the current split between the Gallina chapter and "Extensions of Gallina" is artificial.": 
If the Gallina chapter is the kernel-only stuff and only (some of) the elaborations are the Extensions chapter, perhaps this is not the best example.

Wording suggestions:

- "not really accessible because only presented" -> "not really accessible because **they are** only presented"

- "The "Using Coq" part is expected to gain one or two new chapters about" -> "We expect to add one or two new chapters about"

- "and what user gain with it" -> "and what users gain from it"

- "do not use any of the language extensions provided by extraction" - did you mean "elaboration" instead of "extraction"?

- "split on multiple pages" - "split into"

- "its content is spread out in the corresponding sections, next to more practice-oriented" - 
"... spread among related sections, next to more task-oriented ..."

- "(kills the "Addendum" part)." - "(e.g. getting rid of the "Addendum" part)."

- "this will be a non-user-visible change that it will be possible to backport." - "these will be non-user-visible changes that can be backported"

- "will most often represent strict improvements, and we will strive so that it is the case as often as possible." - "will generally be net improvements.  We'll strive to do so as much as possible."


Let's put a bow on this and present it ASAP.

**Comment by Zimmi48** [comment]
>One of the ideas we discussed was to create chapter/section outlines as we begin working on those chapters/sections. These could be done as PRs to facilitate discussion/review and preserved in the source tree. Could just say we _may_ do this--we'll try and see if it's actually helpful.

Absolutely!

>If the Gallina chapter is the kernel-only stuff and only (some of) the elaborations are the Extensions chapter, perhaps this is not the best example.

I feel like this was close to the original idea of these two chapters but:
- some core stuff like records / sections (that used to be considered as non-core) are only presented in the extension chapter;
- some basic elaboration features are only hinted in the Gallina chapter;
- a lot of elaboration features are only hinted in the extensions chapter and get a full chapter later on.

>Wording suggestions

Thanks!

>did you mean "elaboration" instead of "extraction"?

Yes!

**Comment by Zimmi48** [comment]
>Let's put a bow on this and present it ASAP.

I'll ask Matthieu to validate our current version in #2 and then submit it hopefully in the beginning of next week.

**Comment by Zimmi48** [comment]
@jfehrle In answer to this:

> I will update "Extensions of Gallina" next.

I'd like to proceed ASAP with the splitting of long chapters such as this one in smaller files corresponding to sections. Do you think that this could happen before you start working on this chapter?

**Comment by jfehrle** [comment]
Well, you may recall I already did some work on Gallina Extensions, Syntax Extensions and Ltac, so there would be some wasted motion.  And in fact, I'm in the midst of revising Gallina Extensions today.  How about splitting these 3 chapters after I'm done with them?

I expect it will take you a good while to split all the chapters, which means I'd have to set this project aside until you're done.  I'd rather not do that if it can be avoided.

**Comment by jfehrle** [comment]
Shouldn't we discuss chapter outlines before splitting each chapter?  And if so, the chapters would not be split en masse.

**Comment by Zimmi48** [comment]
> How about splitting these 3 chapters after I'm done with them?

Sure, that will wait then.

>  And if so, the chapters would not be split en masse.

Indeed, the plan is absolutely not to split *en masse*. But splitting can happen even if some detailed outlines are not completely final. Indeed, we can split out some sections before others. And by splitting I mean the fact of separating the code into different files, not a user visible change since it is then included in the file it came from with `.. include`.

**Comment by jfehrle** [comment]
Let me know when you put this in as a CEP.  I don't get notifications for new issues by default.

**Comment by jfehrle** [comment]
Aside from Ltac, I think the commands and tactics chapters will have a lot of edits.  That probably covers the bulk of the syntax updating.  I can try to update these chapters before splitting them into multiple files--when do you expect to split them?  I think the updating will go faster from this point because we're resolved a number of issues along the way.

**Comment by Zimmi48** [comment]
Done at `https://github.com/coq/ceps/pull/43`.

**Comment by Zimmi48** [comment]
> when do you expect to split them?

The "splitting" operation is just meant to occur at a time when no one is working on a file. So if you've already started working on a chapter, then it can wait, but if you haven't, then it could occur before.
